File: game/src/credit.cpp
#include "credit.hpp"

// CreditScene implemenation
// private
// public
void CreditScene::init() {
    done_ = false;
    frame_counter = 0;
}
void CreditScene::update() {
    frame_counter++;
    if (IsMouseButtonPressed(0)) {
        done_ = true;
    }
}
void CreditScene::draw() {
    ClearBackground(RAYWHITE);
    DrawText("insert credit here..", GetScreenWidth() / 2,
             GetScreenHeight() / 2, 20, BLACK);
}
void CreditScene::unload() {}
bool CreditScene::done() { return done_; }

File: src/credit.rs
use crate::engine::transition;
use crate::engine::{Scene, HEIGHT, WIDTH};
use macroquad::prelude::*;

static T: u32 = 300;
static mut FRAME_COUNTER: u32 = 0;

pub unsafe fn init() {
    FRAME_COUNTER = 0;
}
pub unsafe fn update() {
    if FRAME_COUNTER > T && is_mouse_button_pressed(MouseButton::Left) {
        transition(Scene::Menu);
    }
    FRAME_COUNTER += 1;
}

pub unsafe fn draw() {
    clear_background(WHITE);
    draw_text("Credits", 40.0, 40.0, 40.0, BLACK);

    if FRAME_COUNTER > T {
        draw_text(
            "left click to continue",
            WIDTH / 2.0,
            HEIGHT / 2.0,
            20.0,
            GRAY,
        )
    }
}
